test/integration/simple_operations_test.lua

```diff
@@ -490,6 +490,193 @@ pgroup.test_intermediate_nullable_fields_update = function(g)
+            input = {'countries', {{3, box.NULL, 'vatican', 825}},
+                {fetch_latest_metadata = true}},
+            need_pre_insert_data = true,
+        },
+        crud_replace_object_many = {
+            operation_name = 'crud.replace_object_many',
+            input = {'countries',
+                {{id=3, bucket_id = box.NULL, name = 'vatican', population = 825}},
+                {fetch_latest_metadata = true}},
+            need_pre_insert_data = true,
+        },
+        crud_get = {
+            operation_name = 'crud.get',
+            input = {'countries', 3, {fetch_latest_metadata = true}},
+            need_pre_insert_data = true,
+        },
+        crud_delete = {
+            operation_name = 'crud.delete',
+            input = {'countries', 3, {fetch_latest_metadata = true}},
+            need_pre_insert_data = false,
+        },
+        crud_upsert = {
+            operation_name = 'crud.upsert',
+            input = {'countries',
+                {3, box.NULL, 'vatican', 825}, {{'+', 'population', 1}},
+                {fetch_latest_metadata = true}},
+            need_pre_insert_data = false,
+        },
+        crud_upsert_object = {
+            operation_name = 'crud.upsert_object',
+            input = {'countries',
+                {id=3, bucket_id = box.NULL, name = 'vatican', population = 825},
+                {{'+', 'population', 1}},
+                {fetch_latest_metadata = true}},
+            need_pre_insert_data = false,
+        },
+        crud_upsert_many = {
+            operation_name = 'crud.upsert_many',
+            input = {'countries',
+            {
+                {{3, box.NULL, 'vatican', 825},
+                {{'+', 'population', 1}}}
+            },
+            {fetch_latest_metadata = true}},
+            need_pre_insert_data = false,
+        },
+        crud_upsert_object_many = {
+            operation_name = 'crud.upsert_object_many',
+            input = {'countries',
+            {
+                {
+                    {id=3, bucket_id = box.NULL, name = 'vatican', population = 825},
+                    {{'+', 'population', 1}}
+                }
+            },
+            {fetch_latest_metadata = true}},
+            need_pre_insert_data = false,
+        },
+        crud_max = {
+            operation_name = 'crud.max',
+            input = {'countries', 'id', {fetch_latest_metadata = true}},
+            need_pre_insert_data = true,
+        },
+        crud_min = {
+            operation_name = 'crud.min',
+            input = {'countries', 'id', {fetch_latest_metadata = true}},
+            need_pre_insert_data = true,
+        },
+        crud_select = {
+            operation_name = 'crud.select',
+            input = {'countries', {}, {fetch_latest_metadata = true}},
+            need_pre_insert_data = true,
+        },
+    }
+
+    for case_name, case in pairs(cases) do
+        local result, err = g.cluster.main_server.net_box:call(
```

Comment:
The test is dishonest here.

1. We create a space with schema version one.
2. Optionally, we insert some data.
3. We perform a migration to schema version two.
4. We perform a crud operation which reloads schema and returns latest metadata.

Any case that happens after the first one already works with schema version two connections and doesn't reload. To make test cases really test migration, we must do something like the following.


1. We drop the space.
2. We create a space with schema version one.
3. We perform some operation to fetch current schema.
4. Optionally, we insert some data.
5. We perform a migration to schema version two.
6. We perform a crud operation which reloads schema and returns latest metadata.

After implementing, we also should try to revert the changes and ensure that tests fail if we do not perform the reload.

Reply:
Implemented `test_gh_236_dml_operation_return_actual_metadata_*` and `test_gh_236_dml_operation_return_outdated_metadata_without_bugfix_*`